### DISCUSSION TITLE
In branch protection settings, show the status checks that have occurred recently

### DISCUSSION THREAD

**Discussion by Vannevelj** [community/community#16142] [Pull Requests]
Why do I have to search in this input box? I don't know the names of each check by heart. Just give me something that lets me select the checks I want to make required.

<img width="816" alt="image" src="https://user-images.githubusercontent.com/2777107/167107291-285b267b-7391-4778-9e6a-7b7b4fa01791.png">

**Comment by lpascal-ledger** [comment]
May I back this ticket and bring my testimony on why this settings section is very annoying.
The current process of selecting status checks is very ineffective:

- figure out what status check name you need (with workflows, it is the value of the `name` field, but as it is an optional field it falls back to the actual key of the job if the `name` is not explicited);
- write a part of the status check name and wait for the interface to figure out what status it is (the workflow job needs to have ran at least one time, or it won't be found);
- select _one_ job - so if you deal with matrix, you repeat this process `N` time. As the interface does not filter statuses already added, there's a good chance you'll select an already selected one and end up with an error message, and have to repeat;
- some time (2-3 months?) ago, the search field were emptied after selecting a status check, so you were good to write the next name. It has been changed, and now _the field keeps the name of the status checks you've just selected_. Besides being nonsensical (as I just added the status, it's the last one I would like to add _again_, and I could not even add it), it means that now you have to manually remove the field content to type the next status check name.

I understand that the vast majority of users does not spend a lot of time on these settings, but some does - in my case, part of my job is to fork stakeholder repositories into my organization's space and add checks: workflows, git flow and **branch protection** on major branches such as `develop` and `master` (which btw needs to be declared in separate branch protection as the `fnmatch` format can't match these two **and only** these two branches), so I do this ten times or so every week, and this really feels painful.

So as proposed by @Vannevelj, it does feel sound to propose **every** (or latests if too many) **available status checks**, **except the ones which are already added in the list**, and on top of that being able to **select several status checks** at a time.
